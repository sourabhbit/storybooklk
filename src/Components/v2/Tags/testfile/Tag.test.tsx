import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import {
	ComponentSizeENUM,
	ThemeENUM,
	TypographyENUM,
} from '../../../../Types/general'
import Tag from '../index'

const props = {
	id: 'tag',
	name: 'SUITS YOU',
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.primary,
	font: TypographyENUM.serif,
}

describe('<Tag />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<Tag {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('tag-wrapper')
		expect(container).toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
