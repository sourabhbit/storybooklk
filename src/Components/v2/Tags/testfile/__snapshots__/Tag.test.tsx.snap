// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> Should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-bdvvtL goAmab"
    data-testid="tag-wrapper"
    id="tag"
  >
    SUITS YOU
  </span>
</DocumentFragment>
`;
